File: Data Structures & Algorithms/palindrome-number/submission-1.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        num = []
        if x < 0: return False

        while x!=0:
            num.append(x%10)
            x //= 10
       
        for i in range(len(num)//2):
            if num[i] != num[len(num)-i-1]:
                return False
        return True

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> unique;
        for (int num: nums) {
           if (unique.count(num) == 0) {
            unique.insert(num);
           } else {
            return true;
           }
        }
        return false;
        
    }
};